std/assembly/util/sort.ts

```diff
@@ -29,7 +29,11 @@ export function COMPARATOR<T>(): (a: T, b: T) => i32 {
     }
   } else if (isString<T>()) {
     return (a: T, b: T): i32 => {
-      if (a === b || a === null || b === null) return 0;
+      if (a === b) return 0;
+      if (isNullable<T>()) {
+        if (a === null) return  1;
+        if (b === null) return -1;
```

Comment:
What is the criteria to sort `null`s to the end?

Reply:
> What is the criteria to sort nulls to the end?

First of all this align behaviour with JavaScript `arr.sort()`. Also it's much more convenient becouse  when we iterate of sorted array and got `null` we could stop and early return from iteration loop

Reply:
So `[null, 'a', null, 'b', 'c', null, null, 'd', null].sort()` in JS will return `["a", "b", "c", "d", null, null, null, null, null]`.
But `['', 'a', '', 'b', 'c', '', '', 'd', ''].sort()` will return `["", "", "", "", "", "a", "b", "c", "d"]`. AssemblyScript with this PR will have same behaviour.

Reply:
The `null` ending up there is merely a side effect of converting `null` to a string in JS, though:

```
> ["m", null, "o"].sort()
[ 'm', null, 'o' ]
```

Reply:
(not against doing it the way you propose, ofc, but would be good to think about and perhaps establish a convention)

Reply:
Yes, because JavaScript cast `null` to `"null"` string. But we agreed that we will not do this

Reply:
My point is that "First of all this align behaviour with JavaScript arr.sort()." is only correct because of the values used in the test :)

Reply:
this convension much closer to sorting floating points with NaN. Where `Float64Array.of(1, NaN, 2, -Infinity, NaN, -0).sort()` will return `[-Infinity, -0, 1, 2, NaN, NaN]`. And NaNs collected in the end of array

Reply:
> My point is that "First of all this align behaviour with JavaScript arr.sort()." is only correct because of the values used in the test :)

Ah, yeah. That's not true. My bad)

Reply:
My intuition tells me that `null` should probably sort to the front, as it is even "less" (somewhat) than an empty string. Wdyt?

Reply:
I guess `null` is more close to `NaN` (for floats) and should interpret in special way as "not a string" and could exclude from custom sorting rule. But I'm ok with your suggestion as well. Anyway we should collect nulls somehow in special way becouse they're prevent sorting usual strings

Reply:
What you prefer? `[null, null, '', 'a']` or `['', 'a', null, null]`? Last one could be more useful if you're iterate over sorted array

Reply:
I guess it boils down to the question whether `null < ""` or `null > ""`, hmm...

Reply:
> In AssemblyScript null < "" and null > "" shouldn't happening. Isn't?

That's right, but `.sort` still somewhat does such a comparison. As such, I am still split on whether `null` should sort to the beginning or the end. When I sort, I would expect nulls to go to the beginning, because these are essentially "nothing", which in my book is less than "anything". I am not sure, though, if sorting them to the end has real-world advantages, for instance where an algorithm could break when encountering the first null, while when sorted to the beginning, it would have to skip each. Seems rather hypothetical to me, but ultimately I don't know.

> Btw Dart with null safety just throw runtime error in this case =)

A mechanism to just disallow containers with a nullable element type is an interesting alternative so we do not ever have to sort `null`s. Do you have an idea how we could implement this at compile time, say by reusing something TS can do (exclude `null` from a type?), for instance by introducing a `nonnullable<T>` or so type builtin?